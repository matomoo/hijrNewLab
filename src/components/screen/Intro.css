@import '../../root.css';

/* postcss will autoprefix the css properties */
.container {
  display: flex;
  height: 100vh;
  align-self: stretch;
  overflow: scroll;

  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
}
.box {
  position: absolute;
  bottom: 40px;
}
.background {
  background: rgb(65, 77, 107);
}
