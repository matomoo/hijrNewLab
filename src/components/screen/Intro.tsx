import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { inject, observer } from 'mobx-react';

import Button from '../shared/Button';

import classNames from 'classnames/bind';
const classes = require('./Intro.css');
const cx = classNames.bind(classes);

@inject('store') @observer
class Intro extends Component<any, any> {
  public render() {
    const { getString } = this.props.store.locale;
    const containerClass = cx({
      container: true,
      background: true,
    });
    return(
      <div className={containerClass}>
        <div className={classes.box}>
          <Button
            onClick={() => this.onClick()}
            white={true}
            txt={getString('BUTTON')}
          />
        </div>
      </div>
    );
  }

  private onClick = () => {
    alert('clicked');
  }
}

export default Intro;
