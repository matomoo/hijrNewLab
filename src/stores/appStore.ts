import { observable, action } from 'mobx';

import { setSessionStorage, getSessionStorage } from '../utils/Functions';
import Localization from '../models/Localization';
import User from '../models/User';

class Store {
  // todo - make variable properties in the class to private and add getter and setter for those
  @observable private _user: User;
  @observable private _locale: Localization;
  @observable private _isDesktop: boolean;
  @observable private _isOpen: boolean;

  constructor() {
    this.user = new User();
    this.locale = new Localization();
    this.isDesktop = false;
    this.isOpen = false;
  }

  public get user(): User {
    return this._user;
  }
  public set user(value: User) {
    this._user = value;
  }

  public get locale(): Localization {
    return this._locale;
  }
  public set locale(value: Localization) {
    this._locale = value;
  }

  public get isDesktop(): boolean {
    return this._isDesktop;
  }
  public set isDesktop(value: boolean) {
    this._isDesktop = value;
  }

  public get isOpen(): boolean {
    return this._isOpen;
  }
  public set isOpen(value: boolean) {
    this._isOpen = value;
  }

  public getString = (param: string) => {
    return this.locale.getString(param);
  }
}

export default Store;
