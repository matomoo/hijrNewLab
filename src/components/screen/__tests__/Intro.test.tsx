import React from 'react';
import { shallow, render } from 'enzyme';

import appStore from '../../../stores/appStore';
import Intro from '../Intro';

// test for the container page in dom
describe('Intro page DOM rendering test', () => {
  const page = shallow(
    <Intro store={appStore} />,
  );

  it('Tab1 page has to be matched to snapshot', () => {
    expect(page).toMatchSnapshot();
  });
});
