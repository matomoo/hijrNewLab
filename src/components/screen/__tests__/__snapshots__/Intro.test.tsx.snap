// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro page DOM rendering test Tab1 page has to be matched to snapshot 1`] = `
<Intro
  store={[Function]}
/>
`;
